src/util.ts

```diff
@@ -727,13 +754,44 @@ export class Util {
     config: MakeRequestConfig,
     callback: BodyResponseCallback
   ): void | Abortable {
+    let autoRetryValue = AUTO_RETRY_DEFAULT;
+    if (
+      config.autoRetry !== undefined &&
+      config.retryOptions?.autoRetry !== undefined
+    ) {
+      throw new ApiError(
+        'autoRetry is deprecated. Use retryOptions.autoRetry instead.'
+      );
+    } else if (config.autoRetry !== undefined) {
+      autoRetryValue = config.autoRetry;
+    } else if (config.retryOptions?.autoRetry !== undefined) {
+      autoRetryValue = config.retryOptions.autoRetry;
+    }
+
+    let maxRetryValue = MAX_RETRY_DEFAULT;
+    if (config.maxRetries && config.retryOptions?.maxRetries) {
+      throw new ApiError(
+        'maxRetries is deprecated. Use retryOptions.maxRetries instead.'
+      );
+    } else if (config.maxRetries) {
+      maxRetryValue = config.maxRetries;
+    } else if (config.retryOptions?.maxRetries) {
+      maxRetryValue = config.retryOptions.maxRetries;
+    }
+
     const options = {
       request: teenyRequest.defaults(requestDefaults),
-      retries: config.autoRetry !== false ? config.maxRetries || 3 : 0,
+      retries: autoRetryValue !== false ? maxRetryValue : 0,
       shouldRetryFn(httpRespMessage: r.Response) {
         const err = util.parseHttpRespMessage(httpRespMessage).err;
+        if (config.retryOptions?.retryableErrorFn) {
+          return err && config.retryOptions?.retryableErrorFn(err);
+        }
         return err && util.shouldRetryRequest(err);
       },
+      maxRetryDelay: config.retryOptions?.maxRetryDelay,
+      retryDelayMultiplier: config.retryOptions?.retryDelayMultiplier,
+      totalTimeout: config.retryOptions?.totalTimeout,
```

Comment:
@stephenplusplus The system test is failing because it is passing `retry-request` undefined for these three values (`maxRetryDelay`, `retryDelayMultiplier`, and `totalTimeout`). Do you think that common should be defining defaults and always passing those defaults here? Or should `retry-request` be using its defaults if it receives an `undefined` value?

Reply:
`retry-request`! PTAL: https://github.com/stephenplusplus/retry-request/pull/37

Reply:
Amazing! Thank you :)

Reply:
Thanks for catching it! The fix is now released in 4.2.2.